library/packages/src/lib/packages/file_conflict_callbacks.rb

```diff
@@ -0,0 +1,185 @@
+        else
+          Yast::Progress.Step(progress)
+        end
+
+        ui = Yast::UI.PollInput unless Yast::Mode.commandline
+        log.info "User input in file conflict progress (#{progress}%): #{ui}" if ui
+
+        ui != :abort && ui != :cancel
+      end
+
+      # Handle the file conflict detection result callback.
+      # Ask to user whether to continue. In the AutoYaST mode an error is reported
+      # but the installation will continue ignoring the confliucts.
+      # @param excluded_packages [Array<String>] packages ignored in the check
+      #   (e.g. not available for check in the download-as-needed mode)
+      # @param conflicts [Array<String>] list of translated descriptions of
+      #   the detected file conflicts
+      # @return [Boolean] true = continue, false = abort
+      def report(excluded_packages, conflicts)
+        log.info "Excluded #{excluded_packages.size} packages in file conflict check"
+        log.debug "Excluded packages: #{excluded_packages.inspect}"
+        log.info "Found #{conflicts.size} conflicts: #{conflicts.join("\n\n")}"
+
+        # just continue installing packages if there is no conflict
+        return true if conflicts.empty?
+
+        # don't ask in autoyast or command line mode, just report/log the issues and continue
+        if Yast::Mode.auto || Yast::Mode.commandline
+          # TRANSLATORS: An error message, %s is the actual list of detected conflicts
+          Yast::Report.Error(_("File conflicts detected, these conflicting files will " \
+            "be overwritten:\n\n%s") % conflicts.join("\n"))
+          return true
+        end
+
+        Yast::UI.OpenDialog(dialog(conflicts))
+
+        begin
+          Yast::UI.SetFocus(Id(:continue))
+          ret = Yast::UI.UserInput
+          log.info "User Input: #{ret}"
+          ret == :continue
+        ensure
+          Yast::UI.CloseDialog
+        end
+      end
+
+      # Handle the file conflict detection finish callback.
+      def finish
+        log.info "File conflict check finished"
+        return if Yast::Mode.commandline
+
+        # finish the opened progress dialog
+        Yast::Progress.Finish unless pkg_installation?
+      end
+
+      # Construct the file conflicts dialog.
+      # @param [Array<String>] conflicts file conflicts reported by libzypp
+      #   (in human readable form)
+      # @return [Term] UI term
+      def dialog(conflicts)
+        button_box = ButtonBox(
+          PushButton(Id(:continue), Opt(:default, :okButton), Yast::Label.ContinueButton),
+          PushButton(Id(:abort), Opt(:cancelButton), Yast::Label.AbortButton)
+        )
+
+        # TRANSLATORS: A popup label, use max. 70 chars per line, use more lines if needed
+        label = _("File conflicts happen when two packages attempt to install\n" \
+          "files with the same name but different contents. If you continue,\n" \
+          "the conflicting files will be replaced losing the previous content.")
```

Comment:
I have shamelessly taken the [text from zypper...](https://github.com/openSUSE/zypper/blob/27980f701378abcb470bbf5f80abec404e9a6c4a/src/callbacks/rpm.h#L416)